packages/windows/generate_compiled_windows_agent.sh

```diff
@@ -105,15 +95,53 @@ main() {
                 help 1
             fi
             ;;
+        "--dont-build-docker")
+            BUILD_DOCKER="no"
+            shift 1
+            ;;
+        "--tag")
+            if [ -n "$2" ]; then
+                DOCKER_TAG="$2"
+                shift 2
+            else
+                help 1
+            fi
+            ;;
+        "--sources")
+            if [ -n "$2" ]; then
+                CUSTOM_CODE_VOL="-v $2:/local-src:Z"
+                shift 2
+            else
+                help 1
+            fi
+            ;;
         *)
             help 1
         esac
     done
 
-    if [[ "$BUILD" != "no" ]]; then
-        generate_compiled_win_agent || exit 1
+    if [ ! -d "${OUTDIR}" ]; then
+        echo "Creating building directory at ${OUTDIR}"
+        mkdir -p ${OUTDIR}
+    fi
+
+    if [ -z "${CUSTOM_CODE_VOL}" ]; then
+        cd ../..
```

Comment:
We could add a TODO: This needs some work. Consider validating the path to the Wazuh directory (e.g., `*/wazuh*/packages/windows`)